superset/db_engine_specs.py

```diff
@@ -117,16 +138,31 @@ class BaseEngineSpec(object):
     max_column_name_length = None
 
     @classmethod
-    def get_time_expr(cls, expr, pdf, time_grain, grain):
+    def get_time_expr(cls, col: ColumnClause, pdf: Optional[str],
```

Comment:
Shouldn't you also provide a default value if `pdf` and `time_grain` are optional? Also in the docstring the `Optional` word is redundant as that's implied for the type hints.

Reply:
It's my understanding that `Optional[str]` is synonymous with `Union[None, str]`, not that it is an optional argument. Good point about the docstring.